src/EventAccumulator.ts

```diff
@@ -81,19 +118,28 @@ export class EventAccumulator implements IEventAccumulator {
    * @param {string} key - event key
    */
   flush (key: string): void {
-    // if event is not in the store, return.
-    if (!(key in this.events)) {
+    const eventStore = this.events.get(key)
+    if (eventStore == null) {
       return
     }
 
-    const { eventData, startTime, timeout } = this.events[key]
+    const { eventData, startTime, timeout } = eventStore
 
     // update duration to ms from start.
     eventData.dur = Date.now() - startTime
     // add event to the async queue.
-    this.metricsService.q.push(['add_event', eventData])
+    this.metricsService.add_event(eventData)
```

Comment:
kinda-nit-kinda-not: calling metricsService directly here will force us to keep two classes up to date instead of having MetricsProvider be the primary abstraction. I'm fine with this for now because countly will probably be our metrics service for a while but we should try to stick to using MetricsProviders.addEvent() method instead.

i.e. constructor for this should take MetricsProvider instance as argument instead of the metricsService instance.